includes/fbbot/class-omise-fbbot-user-service.php

```diff
@@ -0,0 +1,86 @@
+<?php
+defined( 'ABSPATH' ) or die( "No direct script access allowed." );
+
+if ( class_exists( 'Omise_FBBot_User_Service' ) ) {
+  return;
+}
+
+class Omise_FBBot_User_Service {		
+	private function __construct() {
+		// Hide the constructor
+	}
+
+	public static function get_user( $sender_id ) {
+		$user = get_user_by( 'login', $sender_id );
+
+		if ($user) {
+			return $user->data;
+		}
+
+		$fb_profile = self::get_fb_profile( $sender_id );
+		if ( ! $fb_profile ) 
+			return NULL;
+
+		$wp_user = self::create_wp_user( $fb_profile );
```

Comment:
@igroomgrim there are some concerns about create a new user without telling customer.

- Their username will be reserved and they won't be able to know that you are using their fb as a WooCommerce user name.

- Password has been set but no email, means they won't be able to login with that account directly on a real website for forever (because they don't know their pwd and can't reset them).

- And there is no way to tell pwd to a user except FB chat box unless you integrate with WordPress email system

- etc. 😂